node_package/src/RSCClientRoot.ts

```diff
@@ -5,14 +5,20 @@ import * as ReactDOMClient from 'react-dom/client';
 import { createFromReadableStream } from 'react-on-rails-rsc/client';
 import { fetch } from './utils';
 import transformRSCStreamAndReplayConsoleLogs from './transformRSCStreamAndReplayConsoleLogs';
-import { RailsContext, RenderFunction } from './types';
+import { RailsContext, RenderFunction, RSCPayloadChunk } from './types';
 
 const { use } = React;
 
 if (typeof use !== 'function') {
   throw new Error('React.use is not defined. Please ensure you are using React 19 to use server components.');
 }
 
+declare global {
+  interface Window {
+    REACT_ON_RAILS_RSC_PAYLOAD: RSCPayloadChunk[];
```

Comment:
```suggestion
    REACT_ON_RAILS_RSC_PAYLOAD?: RSCPayloadChunk[];
```
if it doesn't complicate the code (I'm a bit surprised ESLint doesn't complain about `if (!window.REACT_ON_RAILS_RSC_PAYLOAD)` at the moment).